### ISSUE TITLE
Duplicate code for ManureTreatment methods

### ISSUE THREAD

**Issue by allisterakun** [RuminantFarmSystems/RuFaS#1365]
Following PR #1354 and #1359:

The `_adjust_accumulated_output()` method exists in `AnaerobicLagoon`, `AnaerobicDigestionAndLagoon`, `SlurryStorageOutdoor`, and `SlurryStorageUnderfloor` as a temporary fix for empty storage manure at the end of each storage time period.

## Expected Behavior

`_adjust_accumulated_output()` should be moved into `BaseManureTreatment`

## Current Behavior

```python
    def _adjust_accumulated_output(
        self, manure_treatment_daily_output: ManureTreatmentDailyOutput
    ) -> ManureTreatmentDailyOutput:
        """
        Adjust the accumulated output by either resetting it or adding the daily output to it.

        The accumulated output will be reset on the first day of every storage time period.

        Parameters
        ----------
        manure_treatment_daily_output : ManureTreatmentDailyOutput
            The daily output from the manure treatment system.

        Returns
        -------
        ManureTreatmentDailyOutput
            The adjusted accumulated output.

        """
```
The above function (exact same code) exists in `AnaerobicLagoon`, `AnaerobicDigestionAndLagoon`, `SlurryStorageOutdoor`, and `SlurryStorageUnderfloor`.

## Possible Solution

1. Add `self.storage_time_period = self.manure_treatment_config.storage_time_period` (`self.storage_time_period = self.manure_treatment_config[1].storage_time_period` for AnaerobicDigestionAndLagoon) in `BaseManureTreatment.__init__()`. 
2. Move `_adjust_accumulated_output()` into `BaseManureTreatment`.

**Alternatively, instead of step 1, it might be easier to pass the variable `storage_time_period` in as an extra argument to the function `_adjust_accumulated_output()`**
## Steps to Reproduce

1.
3.
4.
5.

**Comment by ew3361zh** [comment]
Yea I was wondering about this possibility - might be able to address it in issue #1272.

**Comment by Isgotkowitz** [comment]
Closed by #1272
